### PR TITLE
Use the fetch stack for HTMLMediaElement.

### PR DESCRIPTION
**Pull request by Ms2ger**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13677)

### CODE REVIEW

File: components/script/dom/htmlmediaelement.rs [hunk 1]
```diff
@@ -508,22 +521,41 @@ impl HTMLMediaElement {
...
+
+            let ty = if self.is::<HTMLAudioElement>() {
+                RequestType::Audio
+            } else if self.is::<HTMLVideoElement>() {
+                RequestType::Video
+            } else {
+                unreachable!("Unexpected HTMLMediaElement")
+            };
+
+            let request = RequestInit {
```
Comment (Manishearth) [inline components/script/dom/htmlmediaelement.rs]: This should be https://html.spec.whatwg.org/multipage/infrastructure.html#create-a-potential-cors-request ?

Comment (Ms2ger) [inline components/script/dom/htmlmediaelement.rs]: Yes, it is; specialized to the case where we don't pay attention to the crossorigin attribute.

Comment (Manishearth) [inline components/script/dom/htmlmediaelement.rs]: Ah, okay. r=me then

**Comment by KiChjang** [comment]
@bors-servo retry
* #13662

**Comment by jdm** [comment]
@bors-servo: retry
* #13662

**Pull request merged**